From ccedc6249576d7562ee434a9e2a5361a50765dc6 Mon Sep 17 00:00:00 2001
From: Martino Facchin <m.facchin@arduino.cc>
Date: Wed, 18 Mar 2020 10:47:06 +0100
Subject: [PATCH 17/31] Hack: QSPIFlash: also remove _needs_fast_mode for
 macronix quirks

Completes e8674c3fc4c78946b6c7c3f38de23d7773733350

This avoids triggering MBED_ASSERT(_num_status_registers == 3); (only applies in debug mode)
---
 .../storage/blockdevice/COMPONENT_QSPIF/QSPIFBlockDevice.cpp    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/components/storage/blockdevice/COMPONENT_QSPIF/QSPIFBlockDevice.cpp b/components/storage/blockdevice/COMPONENT_QSPIF/QSPIFBlockDevice.cpp
index 904b9f7bfb..61dcb6c478 100644
--- a/components/storage/blockdevice/COMPONENT_QSPIF/QSPIFBlockDevice.cpp
+++ b/components/storage/blockdevice/COMPONENT_QSPIF/QSPIFBlockDevice.cpp
@@ -1058,7 +1058,7 @@ int QSPIFBlockDevice::_handle_vendor_quirks()
             // 2. Require setting a "fast mode" bit in config register 2 to operate at higher clock rates
             // 3. Should never attempt to enable 4-byte addressing (it causes reads and writes to fail)
             tr_debug("Applying quirks for macronix");
-            _needs_fast_mode = true;
+            //_needs_fast_mode = true;
             _read_status_reg_2_inst = QSPIF_INST_RDCR;
             _attempt_4_byte_addressing = false;
             break;
-- 
2.27.0

